Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBGlobalErrorResponseHandler.m

```diff
@@ -0,0 +1,171 @@
+///
+/// Copyright (c) 2016 Dropbox, Inc. All rights reserved.
+///
+
+#import "DBGlobalErrorResponseHandler.h"
+#import <objc/runtime.h>
+
+static DBNetworkErrorResponseBlock s_networkErrorResponseBlock = nil;
+static NSOperationQueue *s_networkErrorQueue;
+
+static NSMutableDictionary<Class, DBRouteErrorResponseBlock> * _Nullable s_routeErrorToResponseBlock;
+static NSMutableDictionary<Class, NSOperationQueue *> * _Nullable s_routeErrorToQueue;
+
+static NSLock *s_lock;
```

Comment:
It's probably better to use `@synchronized(someObject) { ... }` instead of this. `@synchronized()` is reentrant and it behaves better with early returns/exceptions.